project/SparkBuild.scala

```diff
@@ -254,7 +254,8 @@ object SparkBuild extends PomBuild {
         // SPARK-45627 `enum`, `export` and `given` will become keywords in Scala 3,
         // so they are prohibited from being used as variable names in Scala 2.13 to
         // reduce the cost of migration in subsequent versions.
-        "-Wconf:cat=deprecation&msg=it will become a keyword in Scala 3:e"
+        "-Wconf:cat=deprecation&msg=it will become a keyword in Scala 3:e",
+        "-Wconf:cat=deprecation&msg=Passing an explicit array value to a Scala varargs method is deprecated:e"
```

Comment:
In the end, we may not need to add this new compile option.

Reply:
Shall we keep it to avoid other folks adding the case again?

Reply:
If it's not necessary, I'll remove it.

Reply:
removed.

Reply:
> Shall we keep it to avoid other folks adding the case again?

This will cause some memory consumption and performance difference in collection copy, so it is indeed a problem. However, I prefer to wait for a while and see if the related cases increase rapidly again. If so, we can clean them up again and make it a stricter compile check.

Reply:
hmm... I want to confirm again that in Scala 3, this is still just a compilation warning, right?